### PR TITLE
num_traits::Num, AsNaturalNumber, AsRingElement

### PR DESCRIPTION
**Pull request by ycscaly**
WIP: for discussion purposes only

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ rand_core = { version = "0.6.4", optional = true }
 rlp = { version = "0.5", optional = true, default-features = false }
 serdect = { version = "0.2", optional = true, default-features = false }
 zeroize = { version = "1", optional = true,  default-features = false }
+num-traits = "0.2"
```
Comment (tarcieri) [inline Cargo.toml:29]: As previously discussed in #158

```suggestion
num-traits = { version = "0.2", optional = true }
```

File: rust-toolchain [hunk 2]
```diff
@@ -0,0 +1 @@
+nightly
```
Comment (tarcieri) [inline rust-toolchain:1]: This crate supports `stable` rust, so making this `nightly` does not make sense

Comment (ycscaly) [reply rust-toolchain:1]: This was just for intellij, shouldn't have been added tomgit, ill remove it but in any case this doesn't effect the PR I just wanted to put some code here to test the idea and have a discussion around it

**Comment by ycscaly** [comment]
Closing after #215  was closed

**Pull request closed without merging**